AH finder computes inertial AH coords, not inertial Strahlkorper.

markscheel opened the pull request:
## Proposed changes
   
We no longer need to compute the full Strahlkorper in the
inertial frame (when finding the Strahlkorper in a non-inertial frame).  Computing merely the inertial-frame coords
of the Strahlkorper collocation points will be sufficient.
See #4132 for details.

Fixes #4132

### Upgrade instructions

<!-- UPGRADE INSTRUCTIONS -->
The full Inertial-frame Strahlkorper is no longer available after horizon finds.  This is ok, because except for spin direction (see #4164) all observables are scalars should be computed in the frame that the Strahlkorper was found, which is more efficient and more straightforward.
<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

markscheel committed:
AH finder: compute inertial coords, not inertial strahlkorper.

We no longer need to compute the full Strahlkorper in the
inertial frame (when finding the Strahlkorper in a non-inertial frame).
Computing merely the inertial-frame coords
of the Strahlkorper collocation points is now sufficient, and is
easier and cheaper.
See #4132 for details.

markscheel committed:
ObserveCenters: Don't use inertial-frame horizon.

Instead take the average of inertial-frame coordinates.

markscheel commented:
@geoffrey4444 especially please look at changes to ObserveSurfaceData; the coords that are being output have now been changed.  If this is not ok, I can output the inertial coords again by grabbing the correct thing from the DataBox.

geoffrey4444 commented:
> @geoffrey4444 especially please look at changes to ObserveSurfaceData; the coords that are being output have now been changed.  If this is not ok, I can output the inertial coords again by grabbing the correct thing from the DataBox.
> 
> 

We do want inertial frame surface output for visualizing AHs. I'll review when I can but teaching workshop 9:39-4 every day this week. Thanks so much for doing this!

knelli2 commented:
I agree with @geoffrey4444 that we want the AH coords (and frame dependent quantities) in the inertial frame. That way we can make movies of inspirals.

markscheel commented:
@geoffrey4444 @knelli2 Ok, I agree.  I have changed ObserveSurfaceData back to outputting inertial coords (but implemented in a different way).  I squashed because nobody has looked at this in detail yet.

By the way, note that this PR fixes a bug in ObserveSurfaceData:  Previously the coords that were output, "InertialCoordinates_x", "InertialCoordinates_y", and "InertialCoordinates_z", did not correspond to the same points that the scalars in ObserveSurfaceData were evaluated on (unless the map was a mixture of only scaling and translation, in which case the points agreed by coincidence.)

knelli2 commented on src/ApparentHorizons/ObserveCenters.hpp:
```diff
@@ -45,9 +50,10 @@ namespace callbacks {
- * \note Requires StrahlkorperTags::Strahlkorper<Frame::Grid> and
- * StrahlkorperTags::Strahlkorper<Frame::Inertial> be in the DataBox of the
- * InterpolationTarget.
+ * \note Requires StrahlkorperTags::Strahlkorper<Frame::Grid>
+ * and StrahlkorperTags::CartesianCoords<Frame::Inertial> and
+ * StrahlkorperGr::Tags::AreaElement<Frame::Grid> to be in the DataBox
+ * of the InterpolationTarget.
```
`static_assert` this

knelli2 commented on tests/Unit/ApparentHorizons/Test_ApparentHorizonFinder.cpp:
```diff
@@ -117,30 +119,28 @@ struct TestHorizonFindFailureCallback {
 };
 
 // If we have found the horizon in a non-inertial frame, the the
-// inertial-frame Strahlkorper should also be in the DataBox.  So this
```
Only one "the" in the first line.

knelli2 reviewed (changes requested):
You can squash while you're rebasing.

markscheel commented:
Fixed typo, added static_asserts, squashed, rebased.

markscheel commented:
(Notice also that ObserveCenters hardcodes Grid frame, so it will need to change when we find horizons in the Distorted frame.  But other things will need to change then too, so waiting for a future PR)

markscheel commented:
(oops, messed up my static_asserts. fixed)
